src/android/com/silkimen/http/HttpRequest.java

```diff
@@ -2489,10 +2490,14 @@ protected HttpRequest copy(final InputStream input, final OutputStream output) t
       public HttpRequest run() throws IOException {
         final byte[] buffer = new byte[bufferSize];
         int read;
-        while ((read = input.read(buffer)) != -1) {
-          output.write(buffer, 0, read);
-          totalWritten += read;
-          progress.onUpload(totalWritten, totalSize);
+        try{
+          while ((read = input.read(buffer)) != -1) {
+            output.write(buffer, 0, read);
+            totalWritten += read;
+            progress.onUpload(totalWritten, totalSize);
+          }
+        }catch(EOFException e){
+          e.printStackTrace();
         }
```

Comment:
Should this also be done in the `copy(Reader` that follows? given it has a similar `while ((read = input.read(buffer)) …` loop.

Reply:
Good idea